kernel/src/engine/default/parquet.rs

```diff
@@ -36,16 +36,22 @@ pub struct DefaultParquetHandler<E: TaskExecutor> {
     readahead: usize,
 }
 
-/// Metadata of a data file (typically a parquet file), currently just includes the file metadata
-/// but will expand to include file statistics and other metadata in the future.
+/// Metadata of a data file (typically a parquet file).
+///
+/// Currently just includes the the number of records as statistics, but will expand to include
+/// more statistics and other metadata in the future.
 #[derive(Debug)]
 pub struct DataFileMetadata {
     file_meta: FileMeta,
+    num_records: usize,
```

Comment:
generally prefer `u64` over `usize` for preditability across platforms unless specifically indexing into collections which already 'speak' `usize`. is there such a requirement here or should we just use `u64`?

Reply:
ah. i see arrow does `usize` for record batch `num_rows`. perhaps a quick note here then?

Reply:
Added a note to the field 👍🏼